### ISSUE TITLE
Is there a little bug in function ngx.decode_args？

### ISSUE THREAD

**Issue by YiYaLi** [openresty/lua-nginx-module#1788]
Code:
    local args = ngx.decode_args("a=hello%world")
    for k, v in pairs(args) do
        ngx.say("k = ", k, ", v = ", v)
    end

Result:
    k = a, v = helloworld

I think the result should be:
    k = a, v = hello%world

**Comment by YiYaLi** [comment]
Yeah, I tested it before. The result is different from the character after '%'.

**Comment by chenxuuu** [comment]
随便改了改，应该可以了

**Comment by suikabreaker** [comment]
> 随便改了改，应该可以了

You need to consider the case when src and dst are the same pointer... that's the point of the interface.

#1862
